### PR TITLE
[CORE][VL] Add resource name for TaskResource interface

### PR DESCRIPTION
**Pull request by Yohahaha**
Add new method `resourceName` for TaskResource to help find which resource may has problem when debug.

### CODE REVIEW

**Comment by Yohahaha** [comment]
@zhztheplayer would you help take a look?

**Review by zhztheplayer — DISMISSED** [review]
Thanks!

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java [hunk 1]
```diff
@@ -25,8 +25,11 @@
 
 public abstract class GeneralOutIterator implements AutoCloseable, Serializable {
   protected final AtomicBoolean closed = new AtomicBoolean(false);
+  protected final long handle;
 
-  public GeneralOutIterator() {}
+  public GeneralOutIterator(long handle) {
+    this.handle = handle;
+  }
```
Comment (zhztheplayer) [inline gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java:32]: Let's avoid adding native-related terms like `handle` to `gluten-core` module. The module doesn't see any native stuffs by design.

**Review by zhztheplayer — COMMENTED** [review]
Thanks!

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java (hunk 1 above)
Comment (zhztheplayer) [reply gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java:32]: I think the principle should be documented (gluten-core: non-native, gluten-data: native) in future. And sorry if that didn't look clear to you.

Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java:32]: Make sense for me.

**Comment by Yohahaha** [comment]
This test still failed, but it passed in my local, seems it does not related with this PR...
```
- SPARK-36726: test incorrect Parquet row group file offset *** FAILED ***[0m
  java.lang.UnsupportedOperationException: Type TimestampNTZType not supported.[0m
  at io.glutenproject.expression.ConverterUtils$.getTypeNode(ConverterUtils.scala:272)[0m
  at io.glutenproject.expression.ConverterUtils.getTypeNode(ConverterUtils.scala)[0m
  at io.glutenproject.substrait.rel.LocalFilesNode.buildNamedStruct(LocalFilesNode.java:89)[0m
  at io.glutenproject.substrait.rel.LocalFilesNode.toProtobuf(LocalFilesNode.java:134)[0m
  at io.glutenproject.substrait.rel.ReadRelNode.toProtobuf(ReadRelNode.java:142)[0m
  at io.glutenproject.substrait.rel.AggregateRelNode.toProtobuf(AggregateRelNode.java:88)[0m
  at io.glutenproject.substrait.plan.PlanNode.toProtobuf(PlanNode.java:76)[0m
  at io.glutenproject.backendsapi.velox.IteratorHandler.genFilePartition(IteratorHandler.scala:123)[0m
  at io.glutenproject.execution.WholeStageTransformer.$anonfun$doExecuteColumnar$5(WholeStageTransformer.scala:265)[0m
  at io.glutenproject.execution.WholeStageTransformer.$anonfun$doExecuteColumnar$5$adapted(WholeStageTransformer.scala:262)[0m
```

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java (hunk 1 above)
Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java:32]: Will update later.

**Comment by baibaichen** [comment]
@Yohahaha would you please add more description?

**Review by baibaichen — COMMENTED** [review]
Can you provide a more detailed description? Purpose, how to use it?

**Comment by Yohahaha** [comment]
@baibaichen This patch does not introduce any new concepts, only add a simple method to help identify TaskResource for improve readability.

**Review by baibaichen — APPROVED** [review]
LGTM

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java (hunk 1 above)
Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java]: Done.

File: gluten-data/src/main/java/io/glutenproject/vectorized/ColumnarBatchOutIterator.java [hunk 2]
```diff
@@ -31,6 +31,11 @@ public ColumnarBatchOutIterator(long handle) throws IOException {
     this.handle = handle;
   }
 
+  @Override
+  public long getHandle() {
+    return handle;
+  }
```
Comment (zhztheplayer) [inline gluten-data/src/main/java/io/glutenproject/vectorized/ColumnarBatchOutIterator.java:37]:
```suggestion
  @Override
  public long getId() {
    return handle; // using the handle value as Id
  }
```

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java [hunk 3]
```diff
@@ -47,6 +47,8 @@ public final void close() {
     }
   }
 
+  public abstract long getHandle();
```
Comment (zhztheplayer) [inline gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java:50]: nit: getHandle -> getId

**Review by zhztheplayer — COMMENTED** [review]
Just one nit. Thanks!

File: gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java (hunk 3 above)
Comment (Yohahaha) [reply gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java]: Done.

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**